### PR TITLE
Make firing of ETW events in nativeaot opt-in via the enabled/disabled lib

### PR DESCRIPTION
**Pull request by elinor-fung**
- Switch the pivot for the enabled/disabled functions to `FireEtw*` instead of `EventPipeWriteEvent*`
  - Functions fire both EventPipe and ETW despite the naming (matches coreclr/mono names)
- Moves support for ETW firing (and helpers only used to determine information for tracing) out of the core runtime and into the enabled lib - same opt-in as EventPipe support

The actual implementation still:
- Has separate tracking from that for the EventPipe support
- Uses the .NET Native provider GUIDs (47c3ba0c-77f1-4eb0-8d4d-aef447f16a85, private: 1095638c-8748-4c7a-b39e-baea27b9c589) instead of the normal .NET ones (e13c0d23-ccbc-4e12-931b-d9cc2eee27e4, private: 763FD754-7086-4dfe-95EB-C01A46FAF4CA).

This will enable us to actually reconcile the ETW and EventPipe tracking/contexts/callbacks in nativeaot, which we will need to address https://github.com/dotnet/runtime/issues/88162 and to switch to the generation scripts based on the event manifest instead of checking in modified parts of the output and manually updating.

I'd also like to move some of the files that are no longer part of the core runtime into a subdirectory (of nativeaot/Runtime) - but in a separate change.

### CODE REVIEW

File: src/coreclr/nativeaot/Runtime/EtwEvents.h [hunk 1]
```diff
@@ -833,33 +825,17 @@ TemplateEventDescriptor(REGHANDLE RegHandle, const EVENT_DESCRIPTOR * Descriptor
...
 #define FireEtwGCOptimized_V1(DesiredAllocation, NewAllocation, GenerationNumber, ClrInstanceID)
 #define FireEtwGCPerHeapHistory()
 #define FireEtwGCSettings(SegmentSize, LargeObjectSegmentSize, ServerGC)
-#define FireEtwPinPlugAtGCTime(PlugStart, PlugEnd, GapBeforeSize, ClrInstanceID)
-#define FireEtwPrvDestroyGCHandle(HandleID, ClrInstanceID)
 #define FireEtwPrvGCMarkCards_V1(HeapNum, ClrInstanceID)
 #define FireEtwPrvGCMarkFinalizeQueueRoots_V1(HeapNum, ClrInstanceID)
 #define FireEtwPrvGCMarkHandles_V1(HeapNum, ClrInstanceID)
 #define FireEtwPrvGCMarkStackRoots_V1(HeapNum, ClrInstanceID)
-#define FireEtwPrvSetGCHandle(HandleID, ObjectID, Kind, Generation, AppDomainID, ClrInstanceID)
```
Comment (LakshanF) [inline src/coreclr/nativeaot/Runtime/EtwEvents.h:839]: Is it possible to add some comments on the FireEtw* and FireEtXplat* mix in the !FEATURE_ETW please?

Comment (elinor-fung) [reply src/coreclr/nativeaot/Runtime/EtwEvents.h:839]: The only one actually used is `FireEtwGCSettings`. I'll delete the rest and move that one over to be like the others (FireEtXplat - which actually fires ETW - in here, FireEtw at the disabled/enabled boundary).

**Review by LakshanF — APPROVED** [review]
Thanks for making this change! The clean separation is going to make it much easier, both to add new events and for the change to auto-generated python scripts.

**Comment by elinor-fung** [comment]
Failure is https://github.com/dotnet/runtime/issues/88870.

**Pull request merged**